alto-pathmanager/api/src/main/java/org/opendaylight/alto/ext/helper/PathManagerHelper.java

```diff
@@ -114,4 +126,51 @@ public static boolean isFlowRuleDiff(Flow before, Flow after) {
     }
     return isNullableObjectDiff(before, after);
   }
+
+
+
+  public static boolean isFlowMatch(FlowDesc rule, FlowDesc test) {
+    return (rule==null) ||
+            ((iPv4PrefixMatch(rule.getSrcIp(),test.getSrcIp()))
+                    && (iPv4PrefixMatch(rule.getDstIp(),test.getDstIp()))
+                    && (protocolMatch(rule.getProtocol(),test.getProtocol()))
+                    && (portNumberMatch(rule.getSrcPort(),test.getSrcPort()))
+                    && (portNumberMatch(rule.getDstPort(),test.getDstPort()))
+                    && (macAddressMatch(rule.getSrcMac(),test.getSrcMac()))
+                    && (macAddressMatch(rule.getDstMac(),test.getDstMac()))
+            );
+
+  }
+  private static boolean iPv4PrefixMatch(Ipv4Prefix rule, Ipv4Prefix test) {
+    if(rule ==null)
+      return true;
+    try {
+      int ruleIPv4Address = IPPrefixHelper.iPv4PrefixToIntIPv4Address(rule);
+      int ruleIPv4Mask = IPPrefixHelper.ipv4PrefixToIntIPv4Mask(rule);
+      int testIPv4Address = IPPrefixHelper.iPv4PrefixToIntIPv4Address(test);
+      int testIPv4Mask = IPPrefixHelper.ipv4PrefixToIntIPv4Mask(test);
+      if(ruleIPv4Mask <= testIPv4Mask) {
+        if((testIPv4Address & ruleIPv4Mask)== (ruleIPv4Address & ruleIPv4Mask))
+          return true;
+        return false;
+      }
+      return false;
+    } catch (UnknownHostException e ) {
+      e.printStackTrace();
```

Comment:
Do not use `e.printStackTrace();`, it is not safe. Try to use `LOG.error("YOUR REASON", e);` instead. Check [this example](https://github.com/openalto/alto-experimental/blob/path-updater/alto-pathmanager/impl/src/main/java/org/opendaylight/alto/ext/impl/help/DataStoreHelper.java#L53).